testing/src/main/java/org/conscrypt/MethodFilter.java

```diff
@@ -0,0 +1,216 @@
+            filter.addPredicate(not(new MethodNamePredicate(names)));
+            return this;
+        }
+
+        /** Expect at least {@code size} matching methods, otherwise fail. */
+        public Builder expectSize(int size) {
+            filter.expectedSize = size;
+            return this;
+        }
+
+        public MethodFilter build() {
+            return filter;
+        }
+    }
+
+    // Implements Builder.hasPrefix()
+    private static class MethodNamePrefixPredicate implements Predicate<Method> {
+        private final String prefix;
+
+        public MethodNamePrefixPredicate(String prefix) {
+            this.prefix = prefix;
+        }
+
+        @Override
+        public boolean test(Method method) {
+            return method.getName().startsWith(prefix);
+        }
+    }
+
+    // Implements Builder.named()
+    private static class MethodNamePredicate implements Predicate<Method> {
+        private final List<String> names;
+
+        public MethodNamePredicate(String... names) {
+            this.names = Arrays.asList(names);
+        }
+
+        @Override
+        public boolean test(Method method) {
+            return names.contains(method.getName());
+        }
+    }
+
+    // Returns a new Predicate which returns the inverse of the original
+    private static Predicate<Method> not(final Predicate<Method> original) {
+        return new Predicate<Method>() {
+            final Predicate<Method> predicate = original;
+
+            @Override
+            public boolean test(Method method) {
+                return !predicate.test(method);
+            }
+        };
+    }
+
+    // Implements Builder.hasArg()
+    private static class MethodArgPredicate implements Predicate<Method> {
+        private final int position;
+        private final List<Class<?>> allowedClasses;
+
+        public MethodArgPredicate(int position, Class<?>... classes) {
+            this.position = position;
+            allowedClasses = Arrays.asList(classes);
+        }
+
+        @Override
+        public boolean test(Method method) {
+            Class<?>[] argTypes = method.getParameterTypes();
+            if (argTypes.length > position) {
+                for (Class<?> c : allowedClasses) {
+                    if (argTypes[position] == c) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+    }
+
+    // Implements Builder.hasArgLength()
+    private static class MethodArgLengthPredicate implements Predicate<Method> {
+        private final int length;
+
+        public MethodArgLengthPredicate(int length) {
+            this.length = length;
+        }
+
+        @Override
+        public boolean test(Method method) {
+            return method.getParameterTypes().length == length;
+        }
+    }
+
+    // Implements Builder.notVoid()
+    private static class MethodIsNotVoid implements Predicate<Method> {
```

Comment:
nit: MethodTakesArguments? Void method seems to be confusing to me.

Reply:
Yeah, I was stuck for a name here too. I started with `hasAtLeastOneArg`.  This predicate is gone, but I'll rename to Builder method to your suggestion.

Reply:
(Done)